packages/affiliates/libs/uniswap/models.ts

```diff
@@ -0,0 +1,253 @@
+import assert from "assert";
+import { exists, getPositionKey } from "./utils";
+import { ethers } from "ethers";
+
+type CacheType = {
+  [key: string]: string | number | boolean;
+};
+export const Cache = (table = new Map<string | number, CacheType>()) => {
```

Comment:
It seems like you're using this in places where you're storing a single, well-defined type in the table. Have you considered making this type a generic?

For a function:
```ts
export function Cache<KeyType, ValueType>(table = new Map<KeyType, ValueType>) {
  // build your return type based on KeyType and ValueType defined inside this function.
}
```

For a class (as suggested below):
```ts
class Cache<KeyType, ValueType> extends Map<KeyType, ValueType> {
  // Not sure if you'll have to tell ts the return type here.
  list() {
    return [...this.values()];
  }
}
```

Reply:
yea this is a really good suggestion. I have a version of this where i tried this and seems to be working